### ISSUE TITLE
Windows 10: when using learn.to_fp16() error

### ISSUE THREAD

**Issue by gerardogarciag1** [fastai/fastai#1432]
**Describe the bug**

https://forums.fast.ai/t/windows-10-list-object-has-no-attribute-dtype-when-learn-fit-one-cycle/34161

I'm trying to use when I invoke 
learn.fit_one_cycle(3,1e-2)

I received the following error 
c:\users\gerar\fastai\fastai\torch_core.py in to_half(b)
    176 def to_half(b:Collection[Tensor])->Collection[Tensor]:
    177     "Set the input of batch `b` to half precision if isn't an int type."
--> 178     return [b[0].half(), b[1]] if b[0].dtype != torch.int64 else b
    179 
    180 def bn2float(module:nn.Module)->nn.Module:

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by sgugger** [comment]
Can you tell us a bit more about your code? What data do you have? What is your model? I'm not sure it's linked to Windows specifically.

**Comment by gerardogarciag1** [comment]
I'm playing with the Microsoft Malware prediction dataset 
[https://www.kaggle.com/c/microsoft-malware-prediction](url) 
I selected these fields for my model 
as part of the cat_names
['EngineVersion',
 'AppVersion',
 'AvSigVersion',
 'IsSxsPassiveMode',
 'AVProductStatesIdentifier',
 'AVProductsInstalled',
 'HasTpm',
 'CountryIdentifier',
 'OrganizationIdentifier',
 'GeoNameIdentifier',
 'LocaleEnglishNameIdentifier',
 'Processor',
 'OsBuild',
 'OsSuite',
 'OsPlatformSubRelease',
 'OsBuildLab',
 'SkuEdition',
 'SMode',
 'IeVerIdentifier',
 'SmartScreen',
 'Firewall',
 'UacLuaenable',
 'Census_MDC2FormFactor',
 'Census_OEMNameIdentifier',
 'Census_ProcessorCoreCount',
 'Census_ProcessorManufacturerIdentifier',
 'Census_ProcessorModelIdentifier',
 'Census_PrimaryDiskTotalCapacity',
 'Census_PrimaryDiskTypeName',
 'Census_SystemVolumeTotalCapacity',
 'Census_HasOpticalDiskDrive',
 'Census_TotalPhysicalRAM',
 'Census_ChassisTypeName',
 'Census_InternalPrimaryDiagonalDisplaySizeInInches',
 'Census_InternalPrimaryDisplayResolutionHorizontal',
 'Census_InternalPrimaryDisplayResolutionVertical',
 'Census_PowerPlatformRoleName',
 'Census_InternalBatteryNumberOfCharges',
 'Census_OSVersion',
 'Census_OSArchitecture',
 'Census_OSBranch',
 'Census_OSBuildNumber',
 'Census_OSBuildRevision',
 'Census_OSEdition',
 'Census_OSSkuName',
 'Census_OSInstallTypeName',
 'Census_OSInstallLanguageIdentifier',
 'Census_OSUILocaleIdentifier',
 'Census_OSWUAutoUpdateOptionsName',
 'Census_GenuineStateName',
 'Census_ActivationChannel',
 'Census_FirmwareManufacturerIdentifier',
 'Census_IsSecureBootEnabled',
 'Census_IsTouchEnabled',
 'Wdft_IsGamer',
 'Wdft_RegionIdentifier']

cont_names =[]
target='HasDetection'

Pretty much standard tabular implementation 
`
data = (TabularList.from_df(train, path='.', cat_names=cat_names, cont_names=cont_names, procs=procs)
                           .random_split_by_pct(0.3)
                           .label_from_df(cols=target)
                           .databunch( num_workers=0))

learn = tabular_learner(data, layers=[200,100], metrics=accuracy, emb_drop=0.1)
#learn.model = nn.DataParallel(learn.model, device_ids=[0,1,2,3])    
#learn.model.to("cuda:0")
learn.to_fp16()
learn.fit_one_cycle(3, 1e-2, wd=0.1)

`

**Comment by sgugger** [comment]
Yes, it's linked to tabular data and the model requiring a list of inputs (and not just one input). Will fix soon.
